# Introduction to Annotations, Terms and Definitions
*by Natalia Ivanova*

Artemis -> Natalia's favorite genome viewer

Artemis colors different types of features with different colors.

Annotations is hard and almost no genomes have this information because there is not a lot of experimental information to back it up anyways.

IMG provides the same annotation and provide mandatory features:
- non-coding RNAs
- Protein-coding genes
- CRISPRs

many ORFs, few CDSs

coding sequences are under different selective pressure.

coding regions third position will have high GC contend (why again !?)

transcribed sequences is more repair than not.

---

RNA prediction
co-variance models
Rfam database, INFERNAL

... other tools

pairwise sequence - BEST smith-waterman alignment

**SAME FUNCTION - FULL LENGTH similarity !!!!!!!!**

full length homologs < full length orthologs

Orthology is usually infered by some heuristic approach, best would be using phylogenetics.

Databases to help annotation:
- COGs cluster of orthologous genes
- KEGG: metabolic pathway collections
- TIGRFAM, really good. Reliable annotation.
- Pfam

KEGG and MetaCyc
Example: Ammonia oxidation pathway
